azurerm/internal/services/maintenance/maintenance_configuration_resource.go

```diff
@@ -62,26 +63,75 @@ func resourceArmMaintenanceConfiguration() *pluginsdk.Resource {
 			"scope": {
 				Type:     pluginsdk.TypeString,
 				Optional: true,
-				Default:  string(maintenance.ScopeAll),
+				Default:  "All",
 				ValidateFunc: validation.StringInSlice([]string{
-					string(maintenance.ScopeAll),
+					"All", // All is still accepted by the API
+					string(maintenance.ScopeExtension),
 					string(maintenance.ScopeHost),
-					string(maintenance.ScopeInResource),
-					string(maintenance.ScopeResource),
+					string(maintenance.ScopeInGuestPatch),
+					string(maintenance.ScopeOSImage),
+					string(maintenance.ScopeSQLDB),
+					string(maintenance.ScopeSQLManagedInstance),
 				}, false),
 			},
 
-			// There's a bug in the Azure API where the the key of tags is returned in lower-case
-			// BUG: https://github.com/Azure/azure-rest-api-specs/issues/9075
-			// use custom tags defition here to prevent inputting upper case key
-			"tags": {
-				Type:         pluginsdk.TypeMap,
-				Optional:     true,
-				ValidateFunc: validate.TagsWithLowerCaseKey,
+			"visibility": {
+				Type:     pluginsdk.TypeString,
+				Optional: true,
+				Default:  string(maintenance.VisibilityCustom),
+				ValidateFunc: validation.StringInSlice([]string{
+					string(maintenance.VisibilityCustom),
+					// Creating public configurations doesn't appear to be supported, API returns `Public Maintenance Configuration must set correct properties`
+					// string(maintenance.VisibilityPublic),
+				}, false),
+			},
+
+			"window": {
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"start_date_time": {
+							Type:     pluginsdk.TypeString,
+							Required: true,
+						},
+						"expiration_date_time": {
+							Type:     pluginsdk.TypeString,
+							Optional: true,
+						},
+						"duration": {
+							Type:     pluginsdk.TypeString,
+							Optional: true,
+							ValidateFunc: validation.StringMatch(
+								regexp.MustCompile("^(0[0-9]|1[0-9]|2[0-3]):[0-5][0-9]$"),
+								"duration must match the format HH:mm",
+							),
+						},
+						"time_zone": {
+							Type:         pluginsdk.TypeString,
+							Required:     true,
+							ValidateFunc: validate.MaintenanceTimeZone(),
+						},
+						"recur_every": {
```

Comment:
what format is this in?

Reply:
That's a tricky one...
 According to the [docs](https://docs.microsoft.com/en-us/rest/api/maintenance/maintenance-configurations/create-or-update):

> Rate at which a Maintenance window is expected to recur. The rate can be expressed as daily, weekly, or monthly schedules. Daily schedule are formatted as recurEvery: [Frequency as integer]['Day(s)']. If no frequency is provided, the default frequency is 1. Daily schedule examples are recurEvery: Day, recurEvery: 3Days. Weekly schedule are formatted as recurEvery: [Frequency as integer]['Week(s)'] [Optional comma separated list of weekdays Monday-Sunday]. Weekly schedule examples are recurEvery: 3Weeks, recurEvery: Week Saturday,Sunday. Monthly schedules are formatted as [Frequency as integer]['Month(s)'] [Comma separated list of month days] or [Frequency as integer]['Month(s)'] [Week of Month (First, Second, Third, Fourth, Last)] [Weekday Monday-Sunday]. Monthly schedule examples are recurEvery: Month, recurEvery: 2Months, recurEvery: Month day23,day24, recurEvery: Month Last Sunday, recurEvery: Month Fourth Monday.